### PR TITLE
Pre commit integration

### PR DESCRIPTION
**Pull request by salilmishra23**
For issue #132, added the following hooks:
- check-yaml
- trailing-whitespace
- requirements-txt-fixer
- end-of-file-fixer
- black

We can add tests and mypy too as hooks but can keep that for another PR. 

Let me know if you want to add/remove hooks.

### CODE REVIEW

**Comment by lalitpagaria** [comment]
@salilmishra23 thank you very much for your PR. 

I will review it by tomorrow and share feedback.

File: obsei/sink/base_sink.py [hunk 1]
```diff
@@ -43,10 +43,7 @@ class BaseSink(BaseModel):
 
     @abstractmethod
     def send_data(
-        self,
-        analyzer_responses: List[TextPayload],
-        config: BaseSinkConfig,
-        **kwargs
+        self, analyzer_responses: List[TextPayload], config: BaseSinkConfig, **kwargs
```
Comment (lalitpagaria) [inline obsei/sink/base_sink.py:46]: Bit strange, on my system black do reverse 👀

File: requirements.in [hunk 2]
```diff
@@ -28,4 +28,4 @@ nltk
 gnews
 trafilatura
 python-facebook-api
-
+pre-commit
```
Comment (lalitpagaria) [inline requirements.in:31]: We can setup dev requirement in other PR to handle all development related dependencies like pytest, black, mypy etc

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+      - id: check-yaml
+        exclude: ^(conda/meta.yaml)
+      - id: trailing-whitespace
+      - id: requirements-txt-fixer
+      - id: end-of-file-fixer
+
+  - repo: https://github.com/psf/black
+    rev: 21.6b0
+    hooks:
+      - id: black
```
Comment (lalitpagaria) [inline .pre-commit-config.yaml:14]: Can we add pytest and mypy as well here?

**Review by lalitpagaria — CHANGES_REQUESTED** [review]
@salilmishra23 Thank you for working on it. I have reviewed your PR. It is looking nice I have few comments. Can you please address them

Also can you please add information about how to perform pre-commit check in contribution guideline

File: obsei/sink/base_sink.py (hunk 1 above)
Comment (salilmishra23) [reply obsei/sink/base_sink.py:46]: yeah, I will check it once again

File: requirements.in (hunk 2 above)
Comment (salilmishra23) [reply requirements.in:31]: Okay, I will first send a PR to add development requirements and then I will finish this PR.

File: .pre-commit-config.yaml (hunk 3 above)
Comment (salilmishra23) [reply .pre-commit-config.yaml:14]: Yes, will add that!

File: obsei/sink/base_sink.py (hunk 1 above)
Comment (salilmishra23) [reply obsei/sink/base_sink.py:46]: I'm using the latest black version 21.6b0

Comment (lalitpagaria) [reply obsei/sink/base_sink.py:46]: okay

**Comment by salilmishra23** [comment]
I have added the `mypy` hook for pre-commit. 

Does adding `pytest` as a hook seem feasible at this point of time?  It has to be run after every commit which takes time specially due to downloading of the huge model. It might not be helpful for people making small quick changes. 

Anyways let me know what you think about this. I have no other problem if you want me to add `pytest` hook.

**Comment by lalitpagaria** [comment]
@salilmishra23 You PR is complete functionality wise. Can you please add information about how to perform pre-commit check in contribution guideline. Post it I will approve and merge this PR.

Okay let's skip pytest for now, we can later add when we feel a need for pytest in pre-commit.
> Does adding pytest as a hook seem feasible at this point of time? It has to be run after every commit which takes time specially due to downloading of the huge model. It might not be helpful for people making small quick changes.

**Review by lalitpagaria — APPROVED** [review]
LGTM!
@salilmishra23 Thank you for working on this.

**Pull request merged**